### DISCUSSION TITLE
Questions for Extend MDAnalysis Interoperability Proposal

### DISCUSSION THREAD

**Discussion by lunamorrow** [MDAnalysis/mdanalysis#4553] [GSoC Q&A] [answered]
I am finalising my application to submit later tonight and I just had a few questions before I submit it through the GSoC portal. 

My proposal is to extend MDAnalysis interoperability with OpenBabel. I am planning to take a similar approach to that used by the current converters, with classes for OpenBabelReader, OpenBabelConverter and OpenBabelParser. The Reader will convert an OpenBabel OBMol to a MDAnalysis AtomGroup, and the Converter will work in reverse. While the Parser will create a Topology from an OpenBabel Forcefield. I am planning to develop these in this order (as per above), and for each one I will develop it in parallel with its tests. The Parser will also require methods in Universe including _resolve_formats and convert_to to be updated. I will be doing all of these conversions through MDAnalysis tools, as well as the automatic, Python wrapped OpenBabel package (PyOBabel). Does this seem like the best approach? I want to ensure my proposal is robust and feasible. Any little tips or suggestions for things I should consider but may have missed would be great!

I would greatly appreciate any insights if you have time @hmacdope @yuxuanzhuang @xhgchen, thanks!

**Comment by hmacdope** [comment, marked as answer]
@lunamorrow this all sounds reasonable to me. I would encourage you to write some mocked up code to see how you think it will look in practice. Its a bit close top deadline but  if you follow https://github.com/MDAnalysis/mdanalysis/wiki/GSoC-FAQ#how-should-i-contact-you-for-feedback-on-my-proposal people can give you feedback directly.

**Reply by hmacdope** [reply]
No worries if not enough time.

**Reply by lunamorrow** [reply]
Thanks @hmacdope! I can't believe I didn't see this page already, I must have missed it. I'm worried it might be a bit late, but I'll see if I can get something drawn up. Worst case, it will be great feedback for me to apply to my project if I am accepted. 🤞

**Reply by lunamorrow** [reply]
I have submitted a draft to GSoC per directions on the "how-to" page. I'll update it with my final application later. 🙌

**Reply by lunamorrow** [reply]
Thank you for your help @hmacdope. I have made those suggested revisions and submitted my final application to GSoC! I really really hope I get offered a place.
